### PR TITLE
Fetch AAD authentication info from backend

### PR DESCRIPTION
**Pull request by dehoward**
### Motivation and Context

this PR prepares the frontend for #377 and removes the need for any AAD configuration environment variables.

### Description
- removes `REACT_APP_AUTH_TYPE` and all variables starting with `REACT_APP_AAD_`
- calls the `/authConfig` endpoint when the app first loads and if needed, renders the `MsalProvider` using the fetched config.
- updates workflows and deployment scripts accordingly

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [Contribution Guidelines](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

File: webapp/src/index.tsx
Comment (glahaye) [inline webapp/src/index.tsx:66]: Why have two separate cases, one with null, the other with undefined?

File: webapp/.env.example
Comment (glahaye) [inline webapp/.env.example:2]: Remove this comment as it doesn't apply anymore,

File: webapp/src/index.tsx
Comment (glahaye) [reply webapp/src/index.tsx:66]: Oh, I just saw the code in App.tsx...  Maybe put a comment here too?

File: webapp/src/App.tsx
Comment (glahaye) [inline webapp/src/App.tsx:225]: Maybe "Unable to load authentication info[...]" or "Unable to authenticate user[...]"?

Also, I would remove the "Oops" from all the strings here.

**Review by glahaye — APPROVED** [review]

File: webapp/src/App.tsx
Comment (dehoward) [reply webapp/src/App.tsx]: ah yes, typo, it should be "Unable to **load** ...".

**Review by glahaye — APPROVED** [review]

**Comment by dehoward** [comment]
needs to be tested in Azure before merging

**Review by alliscode — APPROVED** [review]

File: webapp/src/App.tsx
Comment (teresaqhoang) [inline webapp/src/App.tsx:79]: This setup bypasses the ProbeForBackend state altogether unless we hit a specific error case, but we need to hit the ProbeForBackend state at least once so the maintenceCheck can be hit. 

It might make more sense to Load the AuthInfo in the BackendProbe component in this series of events:
1. LoadAuthInfo
2. then Maintenace Check

If both 1 and 2 pass, then you can bypass the /healthz check there (which is how it's setup now where MaintainenceCheck is queried and the /healthz probe is bypassed if /maintencestatus is hit)

**Review by teresaqhoang — CHANGES_REQUESTED** [review]

**Comment by glahaye** [comment]
> needs to be tested in Azure before merging

It worked fine when I tested it on Azure!

File: webapp/src/App.tsx
Comment (dehoward) [reply webapp/src/App.tsx]: thanks for calling out the maintenance check. I've updated `App` to always begin on the `ProbeForBackend` state, can you take another look?

I ran into issues loading the auth info in `BackendProbe`, since the auth info is used to render the `MsalProvider`, which needs to sit above the `App` in the React tree for the hooks to work. I experimented with having the `BackendProbe` be the top-level- component in `index.tsx`, but it led to too much polling and some other weirdness.

File: webapi/Controllers/MaintenanceController.cs
Comment (glahaye) [inline webapi/Controllers/MaintenanceController.cs:56]: Why change this? What about the RequiresUpgrade state?

Comment (glahaye) [reply webapi/Controllers/MaintenanceController.cs:56]: Moving this to other convo

Comment (glahaye) [reply webapi/Controllers/MaintenanceController.cs:56]: Why change this? What about the RequiresUpgrade state?

Also, maybe we could simplify things and make that flag part of ServiceInfo instead of having to make a call for ServiceInfo and another for Maintenance status.

Comment (dehoward) [reply webapi/Controllers/MaintenanceController.cs:56]: because of this comment on the frontend:
image

there is an existing assumption that when an object is returned from the `/maintenanceStatus` API, that the backend is in maintenance mode, but this is not the case when the status is `RequiresUpgrade` (which mine was in and is why I noticed this).

Comment (crickman) [reply webapi/Controllers/MaintenanceController.cs:56]: I think "RequiresUpgrade" is also valid maintenance status.  Not sure how this change affects maintenance data-flow.

File: webapp/src/components/views/BackendProbe.tsx
Comment (teresaqhoang) [inline webapp/src/components/views/BackendProbe.tsx:58]: Should this be returning true here? If the body returns a payload, that's enough to verify that the backend is up so we should not need to probe.

File: webapi/Controllers/MaintenanceController.cs
Comment (teresaqhoang) [reply webapi/Controllers/MaintenanceController.cs:56]: Doesn't the `RequiresUpgrade` state trigger the upgrade automatically?

File: webapi/Controllers/MaintenanceController.cs
Comment (teresaqhoang) [inline webapi/Controllers/MaintenanceController.cs:1]: Can you remove the authInfo dependency injection here? It might get confusing that this service has a dependency on auth when it doesn't, especially now that this call can be made before auth is loaded.

File: webapp/src/libs/auth/AuthHelper.ts
Comment (teresaqhoang) [inline webapp/src/libs/auth/AuthHelper.ts:32]: Did you mean to remove this? Isn't it still referenced as the default in AuthHelper?

File: webapi/Controllers/MaintenanceController.cs
Comment (dehoward) [reply webapi/Controllers/MaintenanceController.cs]: I'm going to leave it as-is for now to not block the PR

Comment (teresaqhoang) [reply webapi/Controllers/MaintenanceController.cs]: We should either revert this change or confirm that other ChatMigrationStatus states, like `RequiresUpgrade` don't have an effect, otherwise, this will cause a regression.

Comment (dehoward) [reply webapi/Controllers/MaintenanceController.cs]: sorry that's what I meant, that I will revert the change.. on the frontend, it doesn't currently make a big difference since the `isMaintenance` flag isn't set

**Review by teresaqhoang — APPROVED** [review]

**Pull request merged**
